experimental/unitsconversion/src/provider.rs

```diff
@@ -0,0 +1,66 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+//
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use std::borrow::Cow;
+
+use icu_provider::{yoke, zerofrom};
+use zerovec::{VarZeroVec, ZeroMap, ZeroVec};
+use Default;
+
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[repr(u8)]
+pub enum ConstantType {
+    /// The constant value is exact.
+    #[default]
+    EXACT = 0,
+
+    /// The constant value is approximate.
+    APPROXIMATE = 1,
+}
+
+type ConstantTypeMask = u8;
+
+#[icu_provider::data_struct(UnitsConstantsV1Maker = "units/constants@1")]
+#[derive(Default, Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_unitsconversion::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct UnitsConstantsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub constants_map: ZeroMap<'data, str, u16>,
+
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub constants_values: ZeroVec<'data, ConstantValue>,
```

Comment:
Why are you splitting the values into a separate field? Are there duplicates?

Reply:
I am now going to put them in strings, and next PR,  I will do the splitting